test/lib/cli/upgrade_command_test.rb

```diff
@@ -46,10 +84,34 @@ module ForemanMaintain
 
     describe 'check' do
       let :command do
-        %w[upgrade check --disable-self-upgrade]
+        %w[upgrade check]
+      end
+
+      it 'run self upgrade if upgrade available for foreman-maintain' do
```

Comment:
This is an example of one of the failing tests that is running the `:pre_upgrade_checks` and then failing cause it tries to start a fake service. So I think you either need to do some additional mocking here or include a line like you have below:

```
UpgradeRunner.any_instance.expects(:run_phase).with(:pre_upgrade_checks)
```